src/Parser/Core/Modules/Spells/BFA/AzeriteTraits/CrystallineCarapace.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+import Analyzer from 'Parser/Core/Analyzer';
+import { calculateAzeriteEffects } from 'common/stats';
+import SPELLS from 'common/SPELLS';
+
+export function crystallineCarapaceStats(combatant) {
+  if(!combatant.hasTrait(SPELLS.CRYSTALLINE_CARAPACE.id)) {
+    return null;
+  }
+  let armor = 0;
+  let thornDamage = 0;
+  for(const rank of combatant.traitsBySpellId[SPELLS.CRYSTALLINE_CARAPACE.id]) {
```

Comment:
Please use [Array.forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) instead of the old for-loops whenever possible.

Reply:
This could actually be a `reduce` by making the default value the object you're returning anyway.